### PR TITLE
feat(prompts): enrich prompts with vault health, graph data, and actionable instructions

### PR DESCRIPTION
**Pull request by aliasunder**
## Summary

- **vault-orientation**: adds vault stats header (note count, untagged, broken links), folder note counts, property adoption rates with low-adoption flags, orphan detection (top 5), and a contextual footer that recommends tools based on findings
- **memory-review**: adds a structural overview section (file count, scope callouts, section entry counts) before the raw content, enhances scope-fit instructions to reference the overview, and adds a coverage analysis step
- **daily-review**: adds outgoing links (with broken-link flags), backlinks, replaces generic "recently modified" with date-filtered activity via `modifiedOnDate`, and adds task extraction / link following / pattern recognition instructions with dynamic step numbering

Data layer: 3 new `SearchIndex` methods (`brokenLinkCount`, `modifiedOnDate`, `vaultStats`) + `bytes` field on `MemoryFileOutline`. 26 new tests (974 total).

## Test plan

- [x] All 974 tests pass (`npm test`)
- [x] TypeScript compiles clean (`npx tsc --noEmit`)
- [ ] Invoke each prompt via an MCP client — verify output reads well, sections are in order, contextual suggestions appear correctly
- [ ] Edge cases: empty vault, vault with no daily note, memory disabled, past-date daily review

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## New Features
* Enhanced **vault-orientation** prompt now displays vault health metrics including orphan detection, broken links, property adoption rates, and folder statistics.
* Updated **memory-review** prompt with new structural overview section and coverage analysis review step.
* Improved **daily-review** prompt with outgoing/incoming link analysis and date-specific activity tracking.

## Documentation
* Updated documentation across multiple pages describing the three guided prompts and their expanded capabilities.

### CODE REVIEW

File: src/vault-mcp/search/search-index.ts [hunk 1]
```diff
@@ -998,6 +1004,65 @@ export const createSearchIndex = (dbPath: string) => {
...
+  const brokenLinkCount = (logger: Logger): number => {
+    const sql = `
+      SELECT COUNT(*) as count
+      FROM links
+      WHERE target NOT IN (SELECT path FROM notes)
+    `
+    const row = db.prepare(sql).get() as { count: number }
+    logger.info("broken link count", { count: row.count })
+    return row.count
+  }
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/search/search-index.ts]: The `(params, logger)` convention applies to functions that accept params. Parameterless queries take just `(logger)` — this matches the existing `listAllTags(logger)` pattern in the same module. Adding an empty `{}` params object would be ceremony with no value.

File: src/vault-mcp/search/search-index.ts [hunk 2]
```diff
@@ -998,6 +1004,65 @@ export const createSearchIndex = (dbPath: string) => {
...
+  /** Lightweight aggregate counts — total notes, untagged notes, notes without
+   *  frontmatter properties. Single SQL to avoid multiple round-trips. */
+  const vaultStats = (logger: Logger): VaultStats => {
+    const sql = `
+      SELECT
+        COUNT(*) as totalNotes,
+        COALESCE(SUM(CASE WHEN tags = '[]' THEN 1 ELSE 0 END), 0) as untaggedNotes,
+        COALESCE(SUM(CASE WHEN properties = '{}' THEN 1 ELSE 0 END), 0) as noPropertiesNotes
+      FROM notes
+    `
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/search/search-index.ts:1071]: Added an inline comment above the SQL in 265640f — explains the conditional aggregation strategy (count all rows, then conditionally count rows matching the empty-JSON sentinel).

File: src/vault-mcp/search/search-index.ts [hunk 3]
```diff
@@ -998,6 +1004,65 @@ export const createSearchIndex = (dbPath: string) => {
...
+  /** Returns notes whose filesystem mtime falls within a calendar date
+   *  (UTC day boundaries). Enables date-specific activity views. */
+  const modifiedOnDate = (
+    params: { date: string; limit?: number },
+    logger: Logger,
+  ): NoteMetadata[] => {
+    const limit = params.limit ?? 50
+    const dayStart = DateTime.fromISO(params.date, { zone: "utc" })
+    const dayEnd = dayStart.plus({ days: 1 })
+
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/search/search-index.ts:1038]: Fixed in 73593e5 — dropped the `{ zone: "utc" }` override so `modifiedOnDate` uses the system timezone (governed by the `TZ` env var), consistent with `getDailyNote` and `DateTime.now()` throughout the codebase.

File: src/vault-mcp/mcp-core/prompt-definitions.ts [hunk 4]
```diff
@@ -158,33 +263,51 @@ export const registerPrompts = (params: {
...
         const recentSection =
           recent.length > 0
             ? recent.map(formatNoteLine).join("\n")
             : "No notes yet."
+        const orphanSection =
+          orphans.length > 0
+            ? [
+                `${orphans.length}${orphans.length >= ORIENTATION_ORPHAN_LIMIT ? "+" : ""} orphan notes (no incoming links):`,
+                ...orphans.map(formatNoteLine),
+              ].join("\n")
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/prompt-definitions.ts:283]: Good catch — fixed in 265640f. Now fetches `limit + 1` and shows "+" only when results were actually truncated.

**Comment by aliasunder** [comment]
@coderabbit review

File: src/vault-mcp/mcp-core/__tests__/prompt-definitions.test.ts [hunk 5]
```diff
@@ -315,6 +317,167 @@ describe("vault-orientation handler", () => {
...
+  it("shows vault stats header with note count", async () => {
+    const { calls } = await setupVault()
+    const handler = findCall(calls, PROMPT_NAMES.VAULT_ORIENTATION)[2]
+    const text = textOf(await handler(fakeExtra))
+
+    expect(text).toContain("## Vault stats")
+    // 2 indexed notes; 3 folders on disk (About Me, Projects, Reference)
+    expect(text).toContain("2 notes across")
+    expect(text).toContain("3 folders")
+  })
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/__tests__/prompt-definitions.test.ts:332]: The exact-output test covers the full string character-for-character — these behavioral tests are meant to be looser checks for specific features. The substring risk ("12 notes across" matching) is theoretical; the exact test catches real regressions.

File: src/vault-mcp/mcp-core/__tests__/prompt-definitions.test.ts [hunk 6]
```diff
@@ -467,6 +671,209 @@ describe("daily-review handler", () => {
...
+      join(vault, "Daily Notes", "2026-06-16.md"),
+      "# 2026-06-16\n\nPlain text, no links.\n",
+      "utf8",
+    )
+    const handler = findCall(calls, PROMPT_NAMES.DAILY_REVIEW)[2]
+    const text = textOf(await handler({ date: "2026-06-16" }, fakeExtra))
+
+    expect(text).toContain("No outgoing links in this daily note.")
+    expect(text).toContain("No other notes link to this daily note.")
+  })
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/__tests__/prompt-definitions.test.ts:801]: Good catch — fixed in d4ad9b0. Now indexes the daily note into the search index so the link-analysis code path is actually exercised.

File: src/vault-mcp/mcp-core/prompt-definitions.ts [hunk 7]
```diff
@@ -158,33 +265,51 @@ export const registerPrompts = (params: {
...
           recent.length > 0
             ? recent.map(formatNoteLine).join("\n")
             : "No notes yet."
+        const orphanSection =
+          orphans.length > 0
+            ? [
+                `${orphans.length}${hasMoreOrphans ? "+" : ""} orphan notes (no incoming links):`,
+                ...orphans.map(formatNoteLine),
+              ].join("\n")
+            : "No orphans found — every note has at least one incoming link."
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/prompt-definitions.ts:284]: The message targets the common case where no folders are excluded (or exclusions match the defaults). Adding "in the checked scope" would confuse most users who never touch `ORPHAN_EXCLUDE_FOLDERS`. The `vault_find_orphans` tool description already explains exclusions for users who want control.

File: src/vault-mcp/mcp-core/prompt-definitions.ts [hunk 8]
```diff
@@ -342,9 +482,10 @@ export const registerPrompts = (params: {
...
-            "2. **Scope-fit.** Note any entry that seems to belong in a different file or section, based on each file's declared scope (respect a scope header if the file states one; don't assume one otherwise).",
+            "2. **Scope-fit.** Using the scopes shown in the Structure section above, note any entry that seems to belong in a different file or section — does the entry match the file's declared Contains/Does NOT contain scope?",
             "3. **Backfill gaps.** Point out durable facts that are implied but not yet captured, and propose them as dated append entries (bullet + target file + section).",
             `4. **Corrections (rare, separate).** Only a fact that is mis-recorded or now genuinely incorrect — not one that simply changed over time — warrants a fix. Prefer an appended dated correction that preserves the old entry (history matters); reserve vault_delete_memory for genuinely wrong facts.`,
+            "5. **Coverage analysis.** What areas of the user's life, work, or preferences are NOT yet represented? Use the file scopes and section names above to identify gaps worth filling.",
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/prompt-definitions.ts:488]: Intentional — the instruction says "reserve vault_delete_memory for genuinely wrong facts", teaching the LLM that the tool exists but should be used sparingly. Without this, agents wouldn't know the tool exists at all. The prompt IS append-only in its proposals — it asks the user to confirm before any write, including deletion.

File: src/vault-mcp/mcp-core/prompt-definitions.ts [hunk 9]
```diff
@@ -405,15 +546,21 @@ export const registerPrompts = (params: {
...
-              { sort_by: "modified", limit: DAILY_RECENT_LIMIT },
-              reqLogger,
-            ),
-          ),
-        ])
+        const daily = await getDailyNote(
+          { vaultPath, date: args.date },
+          reqLogger,
+        )
+        const dateArg = args.date ?? DateTime.now().toISODate()!
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/prompt-definitions.ts]: Good catch — fixed in d4ad9b0. dateArg is now computed once before getDailyNote, so both always target the same calendar day.

File: src/vault-mcp/mcp-core/prompt-definitions.ts [hunk 10]
```diff
@@ -422,10 +569,56 @@ export const registerPrompts = (params: {
...
+                `**Surface durable facts** — any preference, decision, or fact worth remembering long-term — and propose saving it to ${config.memoryDir}/ memory via vault_update_memory (append-with-dates, newest-first). Confirm before writing.`,
+              ]
+            : []),
+          ...(daily.exists
+            ? [
+                "**Task extraction** — identify any incomplete tasks (`- [ ]`) in the daily note. Are any overdue or blocked?",
+                "**Follow the links** — read linked notes (see outgoing links above) for full context on what was referenced today.",
+                "**Pattern recognition** — look for recurring themes, repeated tasks, or persistent concerns across this note and recent activity.",
+              ]
+            : []),
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/mcp-core/prompt-definitions.ts:617]: The LLM naturally adapts — if the note doesn't exist, the prompt already says "create it with vault_write_note" and the follow-up steps are gated on `daily.exists`. Adding conditional text to the Capture step would over-engineer the prompt for a case the LLM handles gracefully.

File: src/vault-mcp/search/__tests__/search-index.test.ts [hunk 11]
```diff
@@ -2018,3 +2019,197 @@ describe("relative links (path from current file)", () => {
...
+    )
+  })
+
+  it("returns notes modified on the given date", () => {
+    const results = index.modifiedOnDate({ date: "2026-06-15" }, logger)
+    const paths = results.map((note) => note.path)
+    expect(paths).toContain("today-note.md")
+    expect(paths).toContain("today-late.md")
+    expect(paths).toHaveLength(2)
+  })
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/search/__tests__/search-index.test.ts]: Good catch — fixed in d4ad9b0. Both tests now use `toEqual` on the exact path array, including sort order.
